### PR TITLE
Add support for custom network in Google Cloud Life Sciences executor

### PR DESCRIPTION
**Pull request by hnawar**
This pull request adds support for specifying a VPC network to be passed to the Life Sciences API described in issue  #1574

### CODE REVIEW

File: modules/nf-google/src/main/nextflow/cloud/google/lifesciences/GoogleLifeSciencesHelper.groovy [hunk 1]
```diff
@@ -186,8 +186,15 @@ class GoogleLifeSciencesHelper {
...
 
+        def network = new Network()
         if( req.usePrivateAddress ) {
-            vm.setNetwork( new Network().setUsePrivateAddress(true) )
+            network.setUsePrivateAddress(true)
+        }
+        if( req.network ) {
+            network.setNetwork(req.network)
+        }
+        if(req.network!=null | req.usePrivateAddress ) {
```
Comment (pditommaso) [inline modules/nf-google/src/main/nextflow/cloud/google/lifesciences/GoogleLifeSciencesHelper.groovy:196]: I think the condition should `req.network || req.usePrivateAddress` (not null it's evaluated to true by groovy and `|` it's bitwise OR)

**Review by pditommaso — CHANGES_REQUESTED** [review]
Hi Hatem, thanks a lot for your time submitting this PR. The only concern I have is that it addresses bot the network name specification *and* the disk type with the introduction of a new diskType directive. 

Regarding the latter, I remember we discussed this in the past and the plan is to add the type as an attribute of the current `disk` directive.  I apologise if still, I haven't made the required changes to accommodate this need, however, I think it would be better to keep this PR only related to the nextflow enhancement.

**Comment by pditommaso** [comment]
I guess this PR was closed by mistake. I'm merging it manually the avoid including the Cost file and fixing a typo in the docs. Thanks for contributing!

**Comment by pditommaso** [comment]
Merged 3563e171

**Comment by hnawar** [comment]
Hi Paolo,
Please don't merge this PR, I've done some further testing and there are cases that it fails.
I will update you again as it passes.

Hatem

**Comment by pditommaso** [comment]
I see, need to roll back the merge than

**Comment by galvinma** [comment]
Hi, was wondering the status of this issue? Is this slated to be rolled out in production soon?

**Comment by mjpworth** [comment]
I am curious about the state of this feature (specifying a specific network). this is a feature we are also interested in - to comply with our IT security directives. 
Thanks.

**Comment by pditommaso** [comment]
@hnawar if you can comment about what's missing to this PR and can try to solve it.

**Comment by hnawar** [comment]
Hi Paolo,
I'm trying to clean up the changes, also I was only updating the network
but not the subnet which caused this to not work properly when manual
subnets are used. I am trying to test the changes, but this is what is
missing
Hatem

**Comment by pditommaso** [comment]
I see. don't want to put pressure just let me know if you can do it or if you want me or somebody else to. give a try.

**Comment by galvinma** [comment]
FWIW I have been using this code for a while to run jobs in a private VPC: https://github.com/nextflow-io/nextflow/pull/1883/files

Only thing missing would be tests around the subnetwork keyword.

**Comment by pditommaso** [comment]
Solved by 171f4aa8

**Pull request closed without merging**